### PR TITLE
add createApiKeyInterceptor

### PR DESCRIPTION
**Pull request by BlinkyStitt**
I'm not sure the process for updating the changelog and cutting a release

### CODE REVIEW

**Comment by BlinkyStitt** [comment]
example usage:

```

const testHub = async (hubRpcEndpoint, ssl) => {
    let interceptors = [createApiKeyInterceptor('YOUR API KEY HERE')];

    let client = ssl
        ? getSSLHubRpcClient(hubRpcEndpoint, {
              interceptors,
          })
        : getInsecureHubRpcClient(hubRpcEndpoint, {
              interceptors,
          });

    let fid = 3;

    let prettyEndpoint = ssl
        ? `https://${hubRpcEndpoint}`
        : `http://${hubRpcEndpoint}`;

    client.$.waitForReady(Date.now() + 5000, async (e) => {
        if (e) {
            console.error(`Failed to connect to ${prettyEndpoint}: ${e}`);
        } else {
            console.log(`Connected to ${prettyEndpoint}`);

            try {
                const castsResult = await client.getCastsByFid({ fid });
                console.log(
                    `${castsResult.value?.messages.length} casts for fid ${fid} from ${prettyEndpoint}`
                );
            } catch (err) {
                console.error(`Error on ${prettyEndpoint}: ${err.message}`);
            }
        }

        client.close();
    });
};
```

**Review by MrKevinOConnell — CHANGES_REQUESTED** [review]
this looks good - one thing, can you add the usage to the README?

File: src/neynar-api/hub-grpc.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import * as grpc from '@grpc/grpc-js';
+
+export function createApiKeyInterceptor(apiKey: string) {
```
Comment (Shreyaschorge) [inline src/neynar-api/hub-grpc.ts:3]: @BlinkyStitt Can we attach this function as a method to the class `NeynarAPIClient` in `src/neynar-api/neynar-api-client.ts`  so devs don't have to pass in API key while calling this method.

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/neynar-api/hub-grpc.ts (hunk 1 above)
Comment (BlinkyStitt) [reply src/neynar-api/hub-grpc.ts:3]: Its not part of Neynar API Client though. Its attached to the hub as an interceptor

**Comment by BlinkyStitt** [comment]
> this looks good - one thing, can you add the usage to the README?

is the main readme the right place? I figured there were some actual docs pages that would probably make more sense

File: src/neynar-api/hub-grpc.ts (hunk 1 above)
Comment (Shreyaschorge) [reply src/neynar-api/hub-grpc.ts:3]: Then we can have a `HubClient` Class that takes in an API, and this method can be attached to that class. So in future if there is anything related to hub client we can attach it to this client

**Comment by BlinkyStitt** [comment]
doing this in hub-monorepo instead

**Pull request closed without merging**